src/forklift/change_detection.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+# * coding: utf8 *
+'''
+change_detection.py
+a module to track changes from change detection tables
+'''
+import logging
+from os import path
+
+import arcpy
+
+from . import config
+from .models import Crate
+
+log = logging.getLogger('forklift')
+hash_fgdb_name = 'changedetection.gdb'
+hash_fgdb = path.join(config.get_config_prop('hashLocation'), hash_fgdb_name)
+hash_table_name = 'TableHashes'
+hash_table = path.join(hash_fgdb, hash_table_name)
+table_name_field = 'table_name'
+hash_field = 'hash'
+
+
+class ChangeDetection(object):
+    '''A class that models data obtained from the change detection tables
+    '''
+    def __init__(self, table_paths, root_folder, hash_table=hash_table):
+        self.hash_table = hash_table
+        if not arcpy.Exists(hash_table):
+            log.info(f'creating change detection table: {hash_table}')
+            arcpy.management.CreateTable(path.dirname(hash_table), path.basename(hash_table))
+            arcpy.management.AddField(hash_table, table_name_field, 'TEXT')
+            arcpy.management.AddField(hash_table, hash_field, 'TEXT')
+
+        self.current_hashes = _get_hashes([path.join(root_folder, table_path) for table_path in table_paths])
+        self.previous_hashes = _get_hashes([hash_table])
+
+    def has_table(self, table_name):
+        '''table_name: string
+        returns a boolean indicating if there is a current hash for the given table
+        '''
+        return table_name.lower() in self.current_hashes
+
+    def has_changed(self, table_name):
+        '''table_name: string
+        returns a boolean indicating if a tables hash has changed
+        '''
+        table_name = table_name.lower()
+
+        if table_name not in self.current_hashes:
+            raise Exception(f'{table_name} not found in current hashes!')
+
+        if table_name not in self.previous_hashes:
+            return True
+
+        return self.current_hashes[table_name] != self.previous_hashes[table_name]
+
+    def update(self, crate):
+        '''crate: Crate
+        updates the hash table with the current hash and truncates and loads the destination data
+
+        returns an updates crate status
+        '''
+        status = Crate.UPDATED
+        if crate.result[0] == Crate.INVALID_DATA:
+            return crate.result
+        elif crate.result[0] == Crate.CREATED:
+            status = Crate.CREATED
+
+        log.info(f'truncating {crate.destination}')
+        arcpy.management.TruncateTable(crate.destination)
+
+        arcpy.management.Append(crate.source, crate.destination, 'TEST')
```

Comment:
No, from the [append docs](https://pro.arcgis.com/en/pro-app/tool-reference/data-management/append.htm):
>If the spatial references of an input and target feature class do not match, the Append tool will project the features in the input feature class to the coordinate system used by the target feature class.

Although, I didn't think about geographic transformations until now. There is an environment setting [`arcpy.env.geographicTransformations`](https://pro.arcgis.com/en/pro-app/tool-reference/environment-settings/geographic-transformations.htm), but I don't see that listed as an environment that the append tool uses. Maybe I need to project to scratch and then append?

Reply:
We do have a [test from someone on slack](https://thespatialcommunity.slack.com/archives/C0A6GMJMQ/p1583257388018300?thread_ts=1583193073.014400&cid=C0A6GMJMQ) that is promising and we could certainly do our own testing. But I'd feel better if @steveoh can get someone from Esri to sign off on the Append tool honoring `arcpy.env.geographicTransformation` at the Dev Summit before we merge this.

- [ ] I also need to set the environment if we decided to move forward with the append solution.

I do think that it's worth pursuing since the alternative would double the amount of data that the process needs to write.

Reply:
Ted is a smart dude, I would trust his test. I can ask at the dev summit as well but I don't think we need to wait. Maybe I'm anxious to see how the change detection works and need to be patient.

Reply:
I submitted feedback for the Esri docs page. That should get us some sort of answer. How about I do my own testing to confirm that the data looks good and then we can give it a try? I'm going out-of-town tomorrow through Monday so maybe we can give it a go on Tuesday of next week. Although, something tells me that I shouldn't be implementing big changes in forklift while you are off eating sushi in Palm Springs. ;)

Reply:
great idea mr davis. that way everyone wins!

Reply:
Here's my test: https://gist.github.com/stdavis/87f4553aa7638927ea342a0e064cd977. It confirms what Ted found.

Reply:
I wonder if the transformation is stored in the describe object?

Reply:
I wouldn't think so since it describes a transformation between coordinate systems and is not specific to a single one. But let me know if you see otherwise. I'm going to push a commit that adds the transformation environment from the crate property...